oemer/dewarp.py

```diff
@@ -1,28 +1,31 @@
 import os
 import pickle
+from typing import List, Tuple, Any, Union
+from typing_extensions import Self
 
 import cv2
 import numpy as np
 import scipy.ndimage
+import matplotlib.pyplot as plt
+from numpy import ndarray
 from scipy.interpolate import interp1d, griddata
 from sklearn.linear_model import LinearRegression
-import matplotlib.pyplot as plt
 
 from oemer.morph import morph_open
-from oemer.utils import get_logger
+from oemer.logger import get_logger
 
 
 logger = get_logger(__name__)
 
 
 class Grid:
-    def __init__(self):
-        self.id: int = None
-        self.bbox: list[int] = None  # XYXY
+    def __init__(self) -> None:
+        self.id: Union[int, None] = None
+        self.bbox: list[int] | Any = None  # XYXY
```

Comment:
-> `self.bbox: Union[Tuple[int, int, int, int], None]`

Reply:
There are also no none checks (e.g. in line 30) so I used : `bbox: BBox = None  # type ignore` as otherwise I would need to investigate and check a lot of code branches for None handling